### PR TITLE
Add icon support to the `<TabItem>` component

### PR DESCRIPTION
**Pull request by HiDeoo**
#### What kind of changes does this PR include?

- Changes to Starlight code

#### Description

- Closes #1453

This PR adds support for a new optional `icon` prop to the `TabItem` component. This prop can be used to add an icon to a tab to make it easier to visually distinguish between tabs.

image

### CODE REVIEW

File: docs/src/content/docs/guides/components.mdx [hunk 1]
```diff
@@ -58,24 +58,32 @@ These components are available from the `@astrojs/starlight/components` package.
 import { Tabs, TabItem } from '@astrojs/starlight/components';
 
 You can display a tabbed interface using the `<Tabs>` and `<TabItem>` components.
-Each `<TabItem>` must have a `label` to display to users.
+Each `<TabItem>` must have a `label` to display to users and can optionally include an `icon` attribute set to the name of [one of Starlight’s built-in icons](#all-icons).
```
Comment (delucis) [inline docs/src/content/docs/guides/components.mdx:61]: Tiny suggestion to break these into two sentences:

```suggestion
Each `<TabItem>` must have a `label` to display to users.
Use the optional `icon` attribute to include one of [Starlight’s built-in icons](#all-icons) next to the label.
```

**Review by delucis — COMMENTED** [review]
This looks fantastic! I have no notes (well, one tiny suggestion, but given how many notes I sometimes bother you with 😅) — I was thrown for a second by the indirection and then remembered how the tabs component works 😁

File: docs/src/content/docs/guides/components.mdx (hunk 1 above)
Comment (HiDeoo) [reply docs/src/content/docs/guides/components.mdx]: Notes are never a bother, always appreciated!

**Review by delucis — APPROVED** [review]
I declare this iconic PR, ready for launch! 🚀

**Pull request merged**